protos/client.proto

```diff
@@ -56,16 +56,17 @@ message ClientStateCurrentResponse {
 message ClientStateListRequest {
     string merkle_root = 1;
     string subtree = 2;
+    string head = 3;
```

Comment:
Is 'head' always a block id?  If so, should this be called 'block_id'?

These should potentially be different messages - ClientStateListByAddressRequest, ClientStateListByBlockIdRequest.